subprojects/scala/src/main/java/org/gradle/api/internal/tasks/scala/ZincScalaCompilerFactory.java

```diff
@@ -195,7 +231,7 @@ private static File findFile(String prefix, ClassPath classpath) {
     private static String getScalaVersion(ClassLoader scalaClassLoader) {
         try {
             Properties props = new Properties();
-            props.load(scalaClassLoader.getResourceAsStream("library.properties"));
+            props.load(scalaClassLoader.getResourceAsStream("compiler.properties"));
```

Comment:
Can you explain this change?
What does it mean for older versions having a `library.properties`?

Reply:
`library.properties` is contained in `scala_library` and `compiler.properties` is in `scala_compiler` artifact. Both need to be on the classpath and it makes more sense to use the latter. In case of Scala 3 we still use 2.13 standard library, so we can't use `library.properties`

We already do the same thing in Bloop, which uses Zinc underneath https://github.com/scalacenter/bloop/blob/64d8184c812178a3ad949ce38dbba0b8cd4fb53a/backend/src/main/scala/sbt/internal/inc/BloopComponentCompiler.scala#L183